http/server/api.go

```diff
@@ -176,16 +174,32 @@ func (s *rulesetAPI) eval(w http.ResponseWriter, r *http.Request, path string) {
 	reghttp.EncodeJSON(w, r, res, http.StatusOK)
 }
 
-// watch watches a prefix for change and returns anything newer.
-func (s *rulesetAPI) watch(w http.ResponseWriter, r *http.Request, prefix string) {
-	events, err := s.rulesets.Watch(r.Context(), prefix, r.URL.Query().Get("revision"))
+// watch is a long polling endpoint that watches a list of paths for change and returns a list of events containing all the changes
+// that happened since the start of the watch.
+// if the revision query param is specified, it returns anything that happened after that revision.
+// If no paths are specificied, it watches any path.
+// The request context can be used to limit the watch period or to cancel any running one.
+func (s *rulesetAPI) watch(w http.ResponseWriter, r *http.Request) {
+	var paths []string
+
+	// an empty body means watch all paths
+	if r.ContentLength > 0 {
+		err := json.NewDecoder(r.Body).Decode(&paths)
```

Comment:
As a rule, I think it's nicer to avoid top level arrays in POST bodies, as it makes it very hard to extend backward compatibly in future versions. How about using an object with a "paths" member instead? This also it makes it clearer to the client that they're specifying paths and not something else.